src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -444,7 +444,7 @@ public function pluck($column, $key = null)
         // If the model has a mutator for the requested column, we will spin through
         // the results and mutate the values so that the mutated version of these
         // columns are returned as you would expect from these Eloquent models.
-        if ($this->model->hasGetMutator($column)) {
+        if ($this->model->hasGetMutator($column) || $this->model->hasCast($column) || in_array($column, $this->model->getDates())) {
```

Comment:
I considered that but decided against it in the end, but happy to make that change if it's appropriate.